lib/charms/filesystem_client/v0/filesystem_info.py

```diff
@@ -217,12 +217,17 @@ def __init__(
     ) -> None:
         if not scheme:
             raise FilesystemInfoError("scheme cannot be empty")
-        if len(hosts) == 0:
+        str_hosts = []
+        for host in hosts:
+            if not host:
+                raise FilesystemInfoError(f"invalid host `{host}`")
+            str_hosts.append(str(host))
+        if len(str_hosts) == 0:
             raise FilesystemInfoError("list of hosts cannot be empty")
 
         # Strictly convert to the required types to avoid passing through weird data.
         object.__setattr__(self, "scheme", str(scheme))
-        object.__setattr__(self, "hosts", [str(host) for host in hosts])
+        object.__setattr__(self, "hosts", str_hosts)
```

Comment:
Is there a particular reason to not use the built-in [`setattr`](https://docs.python.org/3/library/functions.html#setattr) function here?

```suggestion
        setattr(self, "hosts", str_hosts)
```

Reply:
_TIL!_